packages/components/src/vue/vuex-modules/data/lexis.js

```diff
@@ -312,7 +326,61 @@ export default class Lexis extends Module {
       store.commit('lexis/setTreebankInfo', { hasTreebankData: false })
       this._treebankDataItem = null
     }
-    return annotatedHomonyms
+    return wordIDs
+  }
+
+  async getWordQueryData ({ variables, pollInterval = 1000, maxIterations = 10, source, languageId } = {}) {
+    return new Promise((resolve, reject) => {
+      try {
+        this._adapters.wordQuery.observableQuery({
+          variables,
+          // This callback is called every time the GraphQL query is updated, until the query is complete.
+          dataCallback: (data, state, errors) => { // TODO: How the errors object should be handled?
```

Comment:
and here again we have another dataCallBack for the same wordQuery?

Reply:
Yes, we're using a custom `observableQuery` implementation here, but you're right that we probably don't need it. We can use the standard `observableQuery` implementation with the custom callback function. I will update the code to fix that. Thank you for a very good catch!